### PR TITLE
generalize copy_aws_jar and upgrade emr 5.31.0

### PR DESCRIPTION
**Pull request by guoqiaoli1992**
*Issue #, if available:*
Spark container does not support sse-kms by s3a client with current hadoop version.

*Description of changes:*
Upgrade to emr 5.31.0 with Hadoop 2.10

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/integration/sagemaker/test_spark.py [hunk 1]
```diff
@@ -210,6 +210,66 @@ def test_sagemaker_pyspark_sse_s3(role, image_uri, sagemaker_session, region, sa
...
+    waiter.wait(
+        ProcessingJobName=processing_job.job_name,
+        # poll every 15 seconds. timeout after 15 minutes.
+        WaiterConfig={"Delay": 15, "MaxAttempts": 60},
+    )
+    output_contents = S3Downloader.list(output_data_uri, sagemaker_session=sagemaker_session)
+    assert len(output_contents) != 0
+    for s3_object in s3_client.list_objects(Bucket=bucket, Prefix=output_data_uri_prefix)["Contents"]:
+        object_metadata = s3_client.get_object(Bucket=bucket, Key=s3_object["Key"])
+        assert object_metadata["ServerSideEncryption"] == "aws:kms"
```
Comment (andremoeller) [inline test/integration/sagemaker/test_spark.py:270]: Could you assert that `SSEKMSKeyId` is the expected key as well?

File: test/integration/sagemaker/test_spark.py [hunk 2]
```diff
@@ -210,6 +210,66 @@ def test_sagemaker_pyspark_sse_s3(role, image_uri, sagemaker_session, region, sa
...
+    spark.run(
+        submit_app="test/resources/code/python/hello_py_spark/hello_py_spark_app.py",
+        submit_py_files=["test/resources/code/python/hello_py_spark/hello_py_spark_udfs.py"],
+        arguments=["--input", input_data_uri, "--output", output_data_uri],
+        # configuration=configuration
+        configuration={
+            "Classification": "core-site",
+            "Properties": {
+                "fs.s3a.server-side-encryption-algorithm": "SSE-KMS",
+                "fs.s3a.server-side-encryption.key": f"arn:aws:kms:us-west-2:790336243319:key/{kms_key_id}",
```
Comment (andremoeller) [inline test/integration/sagemaker/test_spark.py]: This ARN has the region `us-west-2` -- could you use the `region` pytest fixture in place of this?

Same for the account ID, which we have a fixture for (named `account_id`):

https://github.com/aws/sagemaker-spark-container/blob/e3dc9678a30b95aa367ab9e4cc112e25d131fdf8/test/integration/conftest.py#L37-L39

Could you pass that `account_id` into this test function and substitute the account ID here with the value of `account_id`?

File: test/integration/sagemaker/test_spark.py [hunk 3]
```diff
@@ -210,6 +210,66 @@ def test_sagemaker_pyspark_sse_s3(role, image_uri, sagemaker_session, region, sa
...
+    )
+
+    kms_key_id = None
+    kms_client = sagemaker_session.boto_session.client("kms", region_name=region)
+    for alias in kms_client.list_aliases()["Aliases"]:
+        if "s3" in alias["AliasName"]:
+            kms_key_id = alias["TargetKeyId"]
+
+    if not kms_key_id:
+        raise ValueError("AWS managed s3 kms does not exist")
```
Comment (andremoeller) [inline test/integration/sagemaker/test_spark.py:231]: Could you add a comment explicitly describing what resources need to already exist in the account for this code to work (the key, the alias, assumptions about names)?

File: test/integration/sagemaker/test_spark.py [hunk 4]
```diff
@@ -210,6 +210,66 @@ def test_sagemaker_pyspark_sse_s3(role, image_uri, sagemaker_session, region, sa
...
+        body = data.read()
+        s3_client.put_object(
+            Body=body, Bucket=bucket, Key=input_data_key, ServerSideEncryption="aws:kms", SSEKMSKeyId=kms_key_id
+        )
+
+    spark.run(
+        submit_app="test/resources/code/python/hello_py_spark/hello_py_spark_app.py",
+        submit_py_files=["test/resources/code/python/hello_py_spark/hello_py_spark_udfs.py"],
+        arguments=["--input", input_data_uri, "--output", output_data_uri],
+        # configuration=configuration
```
Comment (andremoeller) [inline test/integration/sagemaker/test_spark.py:250]: (Nit -- commented out line)

**Review by andremoeller — COMMENTED** [review]
A few notes while this is in progress

File: test/integration/sagemaker/test_spark.py (hunk 2 above)
Comment (guoqiaoli1992) [reply test/integration/sagemaker/test_spark.py]: yep. will do.

**Pull request merged**